mne/io/eeglab/eeglab.py

```diff
@@ -226,6 +233,12 @@ class RawEEGLAB(_BaseRaw):
         Names or indices of channels that should be designated
         EOG channels. If 'auto', the channel names containing
         ``EOG`` or ``EYE`` are used. Defaults to empty tuple.
+    event_id : dict
+        By default, events are read from the input file by dropping every
+        non-integer part of events containing integers, and completely
+        dropping any events without integer parts. If non-integer events
+        should be read, this should be a dict mapping from their names to
+        integers, e.g. dict(fmri_scan_onset=199, recording_start=255).
```

Comment:
Does not seem pragmatic for the typical case. A *very* common case is 3 or 4 string-only or mostly-string events that are unimportant, such as "start recording" or whatever, and a bunch of "S1", "S14" etc.
And a lot of code has stuff like "S  1", "S 10", "S100", which is different non-integer parts, but would be most predictably mapped to 1, 10, and 100.

Also, that would be much more complex code wouldn't it be?